### PR TITLE
fix: combine duplicate Vary headers into single header

### LINKED ISSUE #8847: HTTP header VARY doubled
### Which OpenObserve functionalities are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

HTTP header VARY is doubled:

```
Vary   Origin, Access-Control-Request-Method, Access-Control-Request-Headers
Vary   accept-encoding
```

This behavior triggers Suricata alerts in our environment.

[https://www.rfc-editor.org/rfc/rfc9110#section-5.3](https://www.rfc-editor.org/rfc/rfc9110#section-5.3)

This seems to be NOT an RFC violation, as this header has coma separated value format and MAY be doubled. But for caching consistency across multiple reverse proxies as Nginx + Cloudflare it is advisable to combine values in one header.

Image

### Please provide a link to a minimal reproduction of the bug

N/A

### Please provide the exception or error you saw

```true
N/A
```

### Please provide the version you discovered this bug in (check about page for version information)

```true
Version: v0.15.1
Platform: enterprise
Commit Hash: 80428927172ea157098a4b29280c54629c3ed932
Build Date: 2025-09-04T12:56:46Z
OS: linux / docker
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ByteBaker**
### **User description**
The compression middleware was appending a separate Vary: accept-encoding header instead of combining it with existing CORS Vary headers. This caused Suricata alerts and caching inconsistencies.

Now all Vary values are collected and combined into a single header with case-insensitive deduplication.

Fixes #8847

___

### **PR Type**
Bug fix, Tests

___

### **Description**
- Combine duplicate Vary headers into one

- Deduplicate values case-insensitively

- Preserve existing CORS Vary values

- Add tests for header consolidation

___

### Diagram Walkthrough

```mermaid
flowchart LR
  Encoder["encoding::update_head"]
  HeadersOld["Multiple Vary headers"]
  Collect["Collect + split + trim"]
  Dedup["Case-insensitive check for accept-encoding"]
  Combine["Join into single Vary header"]
  Tests["Unit tests for scenarios"]

  Encoder -- "reads existing Vary" --> HeadersOld
  HeadersOld -- "values extracted" --> Collect
  Collect -- "ensure accept-encoding present" --> Dedup
  Dedup -- "remove old, insert single" --> Combine
  Encoder -- "validated by" --> Tests
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>encoder.rs</strong><dd><code>Consolidate and deduplicate Vary header handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/handler/http/router/middlewares/encoding/encoder.rs

<ul><li>Replace appending Vary with consolidation logic<br> <li> Collect, split, trim, and deduplicate Vary values<br> <li> Ensure single combined Vary header is inserted<br> <li> Add tests for multiple/CORS/duplicate cases</ul>

</details>

  </td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8848/files#diff-1825b00f221539b371b9fe8aa89f9cbc1cfb94f3b731f053e64c51fb5c5934cd">+78/-3</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: src/handler/http/router/middlewares/encoding/encoder.rs [hunk 1]
```diff
@@ -281,8 +281,32 @@ where
...
+        .iter()
+        .any(|v| v.eq_ignore_ascii_case("accept-encoding"))
+    {
+        vary_values.push("accept-encoding".to_string());
+    }
+
+    // Remove all existing Vary headers and insert a single combined one
+    head.headers_mut().remove(header::VARY);
+    let combined = vary_values.join(", ");
+    if let Ok(header_value) = HeaderValue::from_str(&combined) {
```
Comment (uddhavdave) [inline src/handler/http/router/middlewares/encoding/encoder.rs:307]: can you add a log here in case `from_str` fails?
